lib/Injections/NetworkAgent.js

```diff
@@ -0,0 +1,300 @@
+
+      result[key] = '' + value;
+      return result;
+    }, {});
+  }
+
+  function restoreHeadersText(response) {
+    var headers = response.rawHeaders;
+    var headersPrefix = [
+      'HTTP/' + response.httpVersion,
+      response.statusCode,
+      response.statusMessage
+    ].join(' ');
+
+    var joinedHeaders = [headersPrefix];
+    while (headers.length)
+      joinedHeaders.push(headers.splice(-2).join(': '));
+
+    return joinedHeaders.join('\r\n') + '\r\n\r\n';
+  }
+
+  function getMimeType(response) {
+    if (typeof response.headers['content-type'] != 'string')
+      return 'text/plain';
+
+    return response.headers['content-type'].split(';')[0];
+  }
+
+  function constructRequestInfo(request) {
+    var requestId = request.__inspector_ID__;
+    var documentURL = request.__inspector_documentURL__;
+    var timestamp = request.__inspector_timing__.json.requestTime;
+
+    return {
+      requestId: requestId,
+      loaderId: process.pid + '',
+      documentURL: documentURL,
+      type: 'XHR',
+      request: {
+        headers: renderHeaders(request),
+        method: request.method,
+        postData: '',
+        url: documentURL
+      },
+      timestamp: timestamp,
+      initiator: {
+        stackTrace: getStackTrace(),
+        type: 'script'
+      }
+    };
+  }
+
+  function constructResponseInfo(response) {
+    var request = response.req;
+
+    return {
+      requestId: request.__inspector_ID__,
+      loaderId: process.pid + '',
+      timestamp: timestamp(),
+      type: 'XHR',
+      response: {
+        url: request.__inspector_documentURL__,
+        status: response.statusCode,
+        statusText: response.statusMessage,
+        headers: getCorrectHeaders(response),
+        mimeType: getMimeType(response),
+        connectionReused: request.shouldKeepAlive,
+        connectionId: request.connection.__inspector_ID__,
+        encodedDataLength: -1,
+        fromDiskCache: false,
+        fromServiceWorker: false,
+        timing: request.__inspector_timing__.json,
+        headersText: restoreHeadersText(response),
+        requestHeaders: renderHeaders(request),
+        requestHeadersText: request._header
+      }
+    };
+  }
+
+  debug.registerEvent('Network.requestWillBeSent');
+  debug.registerEvent('Network.responseReceived');
+  debug.registerEvent('Network._responseReceived');
+  debug.registerEvent('Network.dataReceived');
+  debug.registerEvent('Network._dataReceived');
+  debug.registerEvent('Network.loadingFinished');
+  debug.registerEvent('Network._loadingFinished');
+
+  wrapHttpRequests();
+  debug.once('close', unwrapHttpRequests);
+
+  function wrapHttpRequests() {
+    var lastRequestId = 0,
+        lastConnectionId = 0;
+
+    function handleHttpRequest() {
+      var timing = new Timing();
+
+      this.__inspector_timing__ = timing;
+      this.__inspector_ID__ = '' + lastRequestId++;
+      this.__inspector_documentURL__ = this._headers.host + this.path;
```

Comment:
This seems to be missing the protocol? Also, as mentioned in the other PR - this doesn't work when forging host headers or setting a port. Especially the latter part is fairly common when talking to internal services. Example:

```js
'use strict';

var http = require('http');

var deferToConnect = http.ClientRequest.prototype._deferToConnect;

http.ClientRequest.prototype._deferToConnect = function defer() {
  // This URL would load just fine
  console.log('Actually not %s', this._headers.host + this.path);
  return deferToConnect.apply(this, arguments);
};

var req = http.get({
  host: '10.255.255.1', port: 8080, headers: { Host: 'api.reddit.com' } // This hangs indefinitely
}, function(res) {});
```